examples/asr/emformer_rnnt/common.py

```diff
@@ -27,11 +28,13 @@ def piecewise_linear_log(x):
     return x
 
 
-def batch_by_token_count(idx_target_lengths, token_limit):
+def batch_by_token_count(idx_target_lengths, token_limit, max_len=None):
     batches = []
     current_batch = []
     current_token_count = 0
     for idx, target_length in idx_target_lengths:
+        if max_len is not None and target_length > max_len:
+            continue
```

Comment:
given the responsibilities the name `batch_by_token_count` implies, this logic would be better placed elsewhere. how about doing it in `CustomDataset`?

Reply:
sounds good, will move it.